05/app.js

```diff
@@ -9,27 +9,73 @@ const stats = {
 
 /* tutaj umieść swój kod */
 
+//Twoim zadaniem jest zbudowanie prostego systemu zliczającego klinięcia w linki na stronie. W obecnym rozwiązaniu przeglądarka ma nie przeładowywać strony tylko zliczać klinięcia w konkretny link tj. .link (identyfikowany po atrybucie href) oraz sumować kliknięcia w paragraf tj. .text (identyfikowany po dataset).
+//Zadanie wykonaj w taki sposób, aby nasłuchiwanie było ustawione tylko na elementy <p/>
+
+
+/*musisz napierw podpiać event klik na <p> .
+Potem sprawdzić czy klikany element to <p> czy <a>. przez e.target.
+W zależności od tego pobierasz identyfikator danego elementu zapisyny jako id lub href.
+Wtedy dopiero robisz ikrementację.*/
+
+const pList = document.querySelectorAll('p');
+console.log(pList);
+
+
+const clickCounter = function (e) {
+    e.preventDefault();
+    const id = e.target.dataset.id;
+    if (typeof stats.paragraphs[id] === 'undefined') {
+        stats.paragraphs[id] = 0;
+    }
+    ++stats.paragraphs[id];
+
+
+    pList.forEach(function (el) {
+        el.addEventListener('click', clickCounter);
+    });
+
+}
+
+const linkList = document.querySelectorAll('a');
+
+const getHref = function (e) {
+    e.preventDefault();
+    const href = e.target.getAttribute('href');
+    console.log('href');
+    if (typeof stats.links[href] === 'undefined') {
+        stats.links[href] = 0;
+    }
+    ++stats.links[href];
+}
+
+linkList.forEach(function (e) {
+    e.addEventListener('click', getHref);
+});
+
```

Comment:
Zadania działa częściowo. Gdy kliknę w `<p>` to nic się nie dzieje - pewnie dlatego, że nie ma eventu `click`.
Druga sprawa jest taka, że zadanie miałoby tak wykonane, że zdarzenie klik może być podpięte tylko pod paragraf (a nie również pod `<a>`). Tutaj należałoby to sprawdzić wew. eventu - na jakim elemencie jest on wywołany przez `e.target`.